pennylane/_device.py

```diff
@@ -98,6 +98,15 @@ def _process_shot_sequence(shot_list):
     return total_shots, shot_vector
 
 
+def _get_num_copies(shot_vector):
+    """Helper function to determine the number of copies from a shot vector Sequence(int) or Sequence(ShotTuple)."""
+    if any(isinstance(shot_comp, ShotTuple) for shot_comp in shot_vector):
```

Comment:
is this only called on something returned from `_process_shot_sequence`? In other words, do you always know that if `any(is ShotTuple)`, then `all(is ShotTuple)`?

Reply:
1. Yes
2. As it stands, yes, but I guess it could potentially change. Users are not expected to create sequences with their own `ShotTuple` objects as elements in the sequence.

Reply:
ah! good to know. ok then I'm much less concerned